File: Assignment-1/Question-4/main.cpp
#include <iostream>
#include <vector>
#include <chrono>
#include <algorithm>
using namespace std;

int main(){
    int n = 100;
    int A[n][n];
    int B[n][n];
    int C[n][n];
    fill(*A, *A + n*n, 1);
    fill(*B, *B + n*n, 1);
    chrono::time_point<chrono::system_clock> start, end;
    
    //itterating through the rows of the matrices
    start = chrono::system_clock::now();
    for(int i = 0; i < n; i++){
        for(int j = 0; j < n; j++){
            C[i][j] = A[i][j] + B[i][j];
        }
    }
    end = chrono::system_clock::now();
    chrono::duration<double> elapsed_seconds = end - start;
    cout << "Row order took" << elapsed_seconds.count() << endl;

    //itterating through the coloums of the matrices
    start = chrono::system_clock::now();
    for(int j = 0; j < n; j++){
        for(int i = 0; i < n; i++){
            C[i][j] = A[i][j] + B[i][j];
        }
    }
    end = chrono::system_clock::now();
    elapsed_seconds = end - start;
    cout << "Row order took" << elapsed_seconds.count() << endl;
}

File: Assignment-1/Question-4/main.py
import time

n = 100
while(n <= 51200):
    print(n)
    A = [[1 for _ in range(n)] for _ in range(n)]
    B = [[1 for _ in range(n)] for _ in range(n)]
    C = [[0 for _ in range(n)] for _ in range(n)]
    #print(n)
    # Iterating through the rows of the matrices
    start = time.time()
    #print(start)
    for i in range(n):
        for j in range(n):
            C[i][j] = A[i][j] + B[i][j]
    end = time.time()
    #print(end)
    elapsed_seconds = end - start
    print("Row order took", elapsed_seconds, "seconds")

    # Iterating through the columns of the matrices
    start = time.time()
    #print(start)
    for j in range(n):
        for i in range(n):
            C[i][j] = A[i][j] + B[i][j]
    end = time.time()
    #print(end)
    elapsed_seconds = end - start
    print("Column order took", elapsed_seconds, "seconds")
    n = n*2
    print()